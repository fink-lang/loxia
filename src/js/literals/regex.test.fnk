{describe, it, expect, to_match_snapshot} = import '@fink/jest/test.fnk'
{raw} = import '@fink/std-lib/str.fnk'

{fink2js} = import '../../testing/generate.fnk'



describe 'regex', fn:
  it 'compiles single line', fn:
    expect
      fink2js raw"
        regex = rx'(?<year>\d{4})-(?<month>\d{2})-(?<day>\d{2})'
      "
      to_match_snapshot


  it 'compiles multiline', fn:
    # TODO: loxia should handle this instead of runtime std-lib
    expect
      fink2js raw"
        regex = rx'
          (?<year>\d{4})-    # year part of a date
          (?<month>\d{2})-   # month part of a date
          (?<day>\d{2})      # day part of a date
        '
      "
      to_match_snapshot


  it 'compiles escape char', fn:
    expect
      fink2js raw"
        regex = rx'.+/\#.+\\'
      "
      to_match_snapshot


