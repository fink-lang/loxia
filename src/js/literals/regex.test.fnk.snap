// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`regex compiles escape char 1`] = `
"const regex_0 = rx\`.+/\\\\#.+\\\\\\\\\`;
export const regex = regex_0;"
`;

exports[`regex compiles multiline 1`] = `
"const regex_0 = rx\`(?<year>\\\\d{4})-    # year part of a date
(?<month>\\\\d{2})-   # month part of a date
(?<day>\\\\d{2})      # day part of a date
\`;
export const regex = regex_0;"
`;

exports[`regex compiles single line 1`] = `
"const regex_0 = rx\`(?<year>\\\\d{4})-(?<month>\\\\d{2})-(?<day>\\\\d{2})\`;
export const regex = regex_0;"
`;
